azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -982,18 +1011,25 @@ private void pollExecution() {
           }
         }
       } else if (FlowRunnerManager.this.active) {
-        try {
-          // Todo jamiesjc: check executor capacity before polling from DB
-          final int execId = FlowRunnerManager.this.executorLoader
-              .selectAndUpdateExecution(this.executorId);
-          if (execId != -1) {
-            FlowRunnerManager.logger.info("Submitting flow " + execId);
-            submitFlow(execId);
-            commonMetrics.markDispatchSuccess();
+        FlowRunnerManager.this.preparingFlowCount.getAndIncrement();
```

Comment:
Having the increment for preparingFlowCount here means that it is blocking for the select from the database as well, which is not needed. In general it's better to keep synchronized/blocking code as minimal and as fast as possible, so that more work can be done in parallel, and to minimize the amount of waiting.

Since this code is not called by the original dispatch logic, the original dispatch logic is still not safe (could still end up preparing flows when inactive).

Reply:
updated, thanks